File: 20242Q/poo/laboratorio/tp04/ej6/src/Expression.java
public abstract class Expression {

    public abstract boolean evaluate();

    public Expression not() {
        return new NotExpression(this);
    }

    public Expression and(Expression expr) {
        return new AndExpression(this, expr);
    }

    public Expression or(Expression expr) {
        return new OrExpression(this, expr);
    }
}

File: 20242Q/poo/laboratorio/tp10/ej3/expression.rb
module Expression
  def evaluate
    raise 'Not implemented'
  end

  # !! todas deben tener métodos que invoquen a constructores
  # entonces, agrego:

  def not
    require_relative 'not_expression'
    NotExpression.new(self)
  end

  def and(expression)
    require_relative 'and_expression'
    AndExpression.new(self, expression)
  end

  def or(expression)
    require_relative 'or_expression'
    OrExpression.new(self, expression)
  end
end

